src/actions/commands/actions.ts

```diff
@@ -671,23 +672,27 @@ class CommandMoveHalfPageDown extends CommandEditorScroll {
       to: this.to,
       by: this.by,
       value: timesToRepeat,
-      revealCursor: false,
+      revealCursor: smoothScrolling ? true : false,
       select:
         [ModeName.Visual, ModeName.VisualBlock, ModeName.VisualLine].indexOf(
           vimState.currentMode
         ) >= 0,
     });
 
-    const newFirstLine = editor.visibleRanges[0].start.line;
-    const newPositionLine = newFirstLine + lineOffset;
-    let newPosition;
+    let newPosition: Position;
+    if (!smoothScrolling) {
```

Comment:
nit: mind reversing this logic such that it's `if (smoothScrolling)`. IMO, affirmation always easier to read than negation.